### ISSUE TITLE
Clean up for Walkabout 

### ISSUE THREAD

**Issue by ccoupe** [Shoes3/shoes3#149]
I could have sworn I've told everybody about 3.3.0 and the repo changes but maybe not.

There is a github remote branch, 'federales' which is a place holder. It's Shoes 3.2.25.  The 'master'  branch will continue to be where new things occur. Only if you need to fix a show stopper bug in 3.2.25 would you `git checkout federales`

I've also deleted a few remote branches - console was merged to master a while back. Some others were 6 year old Shoes 2 or 3.0 experiments, they are gone.  More importantly. I'm going to remove the `gtk3` remote branch soon which might be a problem for @passenger94 

The first thing I'm doing for the `walkabout` release 3.3,x is to get zero warning messages when compiling for any platform. It just needs to be done and I've already found a few "what was he thinking of?" head scratchers along with some Ruby (huh?) and thats just with ruby 2.1.6. Tighten up the ship before dealing with 2.2.x

**Comment by passenger94** [comment]
@ccoupe  no problem ! please proceed as you see fit !

**Comment by ccoupe** [comment]
There are other clean up issues that some folks might care about. The recent XCode update deleted my 10.9 sdk files. We are not happy. Now I get  warnings about  using some 10.8.7 deprecated api's. Guess what I'll do to appease the Apple Gods? @BackOrder - we new this moment would arrive.

**Comment by BackOrder** [comment]
What are the implications of this? I am surprised that you updated considering that you are always so careful about NOT updating Shoes libraries. How about backups? Don't you use Timeline or something?

**Comment by ccoupe** [comment]
It means people who want Shoes on OSX are going to need to upgrade their software (and possibly the hardware) to something current.  TimeMachine requires a second disk which is more than I want to spend. Shoes is backup up. Xcode - no.

**Comment by BackOrder** [comment]
Downgrade. :)

Not everybody is upgrading their systems. It sounds like a real problem.

**Comment by ccoupe** [comment]
How do you do that w/o a DVD drive (and disk?). I'm too cheap to buy them too.

**Comment by BackOrder** [comment]
Do you mean Xcode? There is some way to download previous version from Apple. Not exactly sure how though.

**Comment by ccoupe** [comment]
Older Xcode versions can be downloaded from Apple. Will they install in 10.11?  I suspect it won't and if it does it's going to break things in weird ways.  What I need is the tar ball of just the  10.9 sdk tree, similar to what you sent long ago for 10.6. Hell, I might have one in a VM if I could be bothered to look and I'm sure they'll appear in the internets somewhere if not from Apple. 

The issue is that I'm not going to do that 10.6 dance again for 10.7 and then 10.8 and so on as Apple moves the goalposts. I'd rather write better wiki articles than litter cocoa.m up with even more #ifdefs.

**Comment by ccoupe** [comment]
OK! I found these on my backup server (back I did some 10.6 stuff)
```
-rw-r--r-- 1 ccoupe ccoupe 5653921792 Apr 23  2014 OSX-Mavericks.dmg
-rw-r--r-- 1 ccoupe ccoupe 5454309376 May  1  2014 OSX-Mavericks.iso
-rw-r--r-- 1 ccoupe ccoupe   21575668 Nov  9  2013 webmin_1.660_all.deb
-rw-r--r-- 1 ccoupe ccoupe 1723816316 Apr 23  2014 Xcode 4.6.3.dmg
-rw-r--r-- 1 ccoupe ccoupe 2195476273 Apr 23  2014 xcode_5.0.2_gm_seed.dmg
```
FWIW.  I  picked up a 1.5TB usb drive, hooked it up to my raspberry pi2 and with some magic dust sprinkled around it's 500GB TimeCapsule and will soon be a 1TB backup of the linux box(es) and VM's.

**Comment by BackOrder** [comment]
Good boy!

**Comment by ccoupe** [comment]
I did get SDK 10.9 out of that `xcode_5.0.2_gm_seed.dmg ` and rebuilt Shoes OSX against it. Works for me but I'm not on 10.9  - http://walkabout.mvmanila.com/public/shoes/shoes-3.3.0-osx-10.9.tgz

I also managed to backup my Linux  chroots to the raspberry pi2. No small feat because they are root and the pi (and most new linux) don't allow remote root logins

**Comment by ccoupe** [comment]
I still have work to do that belongs here.

**Comment by ccoupe** [comment]
The last part of the cleanup for 3.3 is the Wiki. I'd like to make it more structured or manual like (with a thought to @backorders epub code). There are features of the wiki that are not well known. If two people clone it then their local changes follow the normal git commit/push/pull/merge.
https://help.github.com/articles/adding-and-editing-wiki-pages-locally/

Even less well known (by me) is that a directory/folder hierarchy is possible with a unique sidebar  for each level, according to https://roland.codes/blog/github-wiki-sidebars/ That needs testing. So I will

**Comment by ccoupe** [comment]
More dependency issues to fix before this issue can be closed. And of course it's weird.  I noticed ask_open_file was not working on the raspberry pi(2) with gtk3 - which is based on an old debian   chroot and an old gcc 4.6 cross compiler. The raspberry pi2 runs a version of debian jessie. I could create a new chroot/qemu  for jessie with more recentl/fixed gtk libs (3.2.35 used gtk2) and get or build a newer gcc cross compiler **OR** I could a new rakefile target (pi2) that did the Tight Shoes build on the raspberry pi2. I choose the latter and many hours later-  it did fix the ask_open_file bug

There's still a butt load of gtk2 things in the rakefiles that doesn't need to be there for walkabout. So, while @passenger94 is still working on #54, my particular skills are better spent cleaning up the build and dependencies.
